hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionTool.java

```diff
@@ -97,4 +112,32 @@ private void putAndFlush(int key) throws Exception {
     region.flush(true);
   }
 
+  @Test
+  public void testCompactedFilesArchivedMapRed() throws Exception {
+    for (int i = 0; i < 10; i++) {
+      this.putAndFlush(i);
+    }
+    HStore store = region.getStore(HBaseTestingUtility.fam1);
+    assertEquals(10, store.getStorefilesCount());
+    Path tableDir = CommonFSUtils.getTableDir(rootDir, region.getRegionInfo().getTable());
+    FileSystem fs = store.getFileSystem();
+    String storePath = tableDir + "/" + region.getRegionInfo().getEncodedName() + "/"
+      + Bytes.toString(HBaseTestingUtility.fam1);
+    FileStatus[] regionDirFiles = fs.listStatus(new Path(storePath));
+    assertEquals(10, regionDirFiles.length);
+    String defaultFS = testUtil.getMiniHBaseCluster().getConfiguration().get("fs.defaultFS");
+    testUtil.startMiniMapReduceCluster();
+    try {
+      Configuration config = HBaseConfiguration.create(testUtil.getConfiguration());
+      config.setBoolean(MemStoreLAB.USEMSLAB_KEY, true);
```

Comment:
Do we still need to do this after parameterized?

Reply:
Parameterization is for Server Config. This config is for CompactionTool(Always passing true). Issue happens when mslab is true for Compactiontool. Need to do mslab as false for server, so that ChunkCreator is not created, Otherwise CompactionTool flow was using the Server ChunkCreator instance as test is running in same jvm instance.

Reply:
I do not get the point here...

We will run the test method twice with USEMSLAB_KEY as true and false, so we can cover the scenario where we pass USEMSLAB_KEY as true? If we only want it to be true, we should give this test method a separated test class?

Reply:
Base test is about functional verification of the feature. For this issue, we can add test cases to a new file. I can revert to test code which I have initially pushed. What is your opinion?

Reply:
If you want to add a new testcase, then just introduce a new test class so we do not need to run the test method in parent class twice? This is my very first question...

Reply:
Test code update done. Please review...